### DISCUSSION TITLE
Google Provider: Application "wants additional access to your Google Account"

### DISCUSSION THREAD

**Discussion by fromi** [nextauthjs/next-auth#10970] [Help] [answered]
Since I migrated to next-auth 4, everytime I login with my Google Account I have this page:
![image](https://github.com/nextauthjs/next-auth/assets/6806494/b09b3f45-2acf-4775-b4b0-9d3d6f63854f)

When I try to login with my pro account, I do not have this page displayed!

I found this thread about the issue on Stack Overflow: https://stackoverflow.com/questions/72148057/app-name-wants-additional-access-to-your-google-account
I would like to try to set `include_granted_scopes` to false when calling the Google API, how can I try that using NextAuth Google Provider configuration?

**Comment by k-taro56** [comment]
maybe
https://authjs.dev/reference/core/providers#authorization

```ts
Google({
  authorization:
    params: {
      include_granted_scopes: false
    }
}),
```

**Reply by fromi** [reply]
Thanks, it works for passing the parameters, however it does not resolve my issue, I still have the ugly screen every time I login.
I noticed that I am redirected to `https://accounts.google.com/o/oauth2/v2/` since I migrated to next-auth v4. Previously I was redirected to `https://accounts.google.com/o/oauth2/`. Maybe there is a bug on the google auth v2 API, or the way it in integrated?

**Reply by k-taro56** [reply]
If you need to revert the `Provider` behavior back to next-auth v4, please change the following.
Note: This is not the best way.

```ts
Google({
  type: "oauth",
  issuer: "",
  wellKnown: "https://accounts.google.com/.well-known/openid-configuration",
  authorization: { params: { scope: "openid email profile" } },
  idToken: true,
  checks: ["pkce", "state"],
  profile(profile) {
    return {
      id: profile.sub,
      name: profile.name,
      email: profile.email,
      image: profile.picture,
    }
  },
  style: {
    brandColor: "#1a73e8",
  },
}),
```

**Reply by fromi** [reply, marked as answer]
Hi, it turn out the knew URL for Google Auth (api/v2) is giving the strange warning message but only for localhost URL (probably because it cannot be verified).
Once deployed online, everything works fine.
